examples/dmd/generic_csv.cpp

```diff
@@ -155,190 +159,234 @@ int main(int argc, char *argv[])
-                    }
-                    else
-                    {
-                        dmd[curr_window] = new CAROM::DMD(dim, dtc);
-                    }
                     dmd[curr_window]->takeSample(sample, tval);
                 }
             }
-        }
 
-        if (myid == 0)
-        {
-            cout << "Loaded " << num_train_snap[idx_dataset] << " samples for " << par_dir << "." << endl;
-        }
-    }
-
-    for (int window = 0; window < numWindows; ++window)
-    {
-        if (rdim != -1)
-        {
             if (myid == 0)
             {
-                cout << "Creating DMD model #" << window << " with rdim: " << rdim << endl;
+                cout << "Loaded " << num_train_snap[idx_dataset] << " samples for " << par_dir << "." << endl;
             }
-            dmd[window]->train(rdim);
         }
-        else if (ef != -1)
+
+        for (int window = 0; window < numWindows; ++window)
         {
+            if (rdim != -1)
+            {
+                if (myid == 0)
+                {
+                    cout << "Creating DMD model #" << window << " with rdim: " << rdim << endl;
+                }
+                dmd[window]->train(rdim);
+            }
+            else if (ef != -1)
+            {
+                if (myid == 0)
+                {
+                    cout << "Creating DMD model #" << window << " with energy fraction: " << ef << endl;
+                }
+                dmd[window]->train(ef);
+            }
+            dmd[window]->save(outputPath + "/window" + to_string(window)); 
             if (myid == 0)
             {
-                cout << "Creating DMD model #" << window << " with energy fraction: " << ef << endl;
+                dmd[window]->summary(outputPath + "/window" + to_string(window)); 
             }
-            dmd[window]->train(ef);
         }
-        dmd[window]->summary(outputPath + "/window" + to_string(window)); 
-    }
 
-    dmd_training_timer.Stop();
-
-    CAROM::Vector* result = new CAROM::Vector(dim, true);
-    if (ddt > 0.0 && npar == 1) // Assuming indicator value increases with snapshot order. Not true for parametric.
-    {
-        CAROM::AdaptiveDMD* admd = nullptr; 
-        CAROM::Vector* interp_snap = new CAROM::Vector(dim, true);
-        vector<double> interp_error;
+        if (myid == 0 && windowNumSamples < infty)
+        {
+            csv_db->putDoubleVector(string(data_dir) + "/indicator_val.csv", indicator_val, numWindows);
+        }
+        dmd_training_timer.Stop();
 
-        for (int window = 0; window < numWindows; ++window)
+        if (ddt > 0.0 && npar == 1) // TODO: Verify at all training par
         {
-            admd = dynamic_cast<CAROM::AdaptiveDMD*> (dmd[window]);
-            double t_init = dmd[window]->getTimeOffset();
+            CAROM::AdaptiveDMD* admd = nullptr; 
+            CAROM::Vector* interp_snap = new CAROM::Vector(dim, true);
```

Comment:
Can this just be `CAROM::Vector interp_snap(dim, true);` so that it doesn't have to be deleted later?

Reply:
Yes, they are now modified.